Tickr/Tickr/StockManager.swift

```diff
@@ -0,0 +1,238 @@
+     */
+    func parseStockData(stockData: NSArray) {
+        stockArray.removeAll(keepCapacity: false)
+        for stock in stockData {
+            var name: String = ""
+            var symbol: String = ""
+            var price: String = "0"
+            var changeInPercentString: String = "0"
+            var changeInPriceString: String = "0"
+            var changeInPercentStringClean: NSString = "0"
+            var changeInPriceStringClean: NSString = "0"
+            let keys = Stock.SerializationKeys.self
+            
+            //Stock Company Name
+            if let n = stock[keys.name] as? String {
+                name = n
+            }
+            
+            //Stock Symbol
+            if let s = stock[keys.symbol] as? String {
+                symbol = s.uppercaseString
+            }
+            
+            //Stock Price
+            if let p = stock[keys.price] as? String {
+                price = p
+            }
+            
+            //Stock Increase or Decrease as Percentage
+            if let cPercent = stock[keys.changeInPercent] as? String {
+                changeInPercentString = cPercent
+            
+                changeInPercentStringClean = (changeInPercentString as NSString).substringToIndex(changeInPercentString.characters.count-1)
+            }
+            
+            //Stock Increase or Decrease as Price
+            if let cPrice = stock[keys.changeInPrice] as? String {
+                changeInPriceString = cPrice
+            
+                changeInPriceStringClean = (changeInPriceString as NSString).substringToIndex(changeInPriceString.characters.count-1)
+            }
+            
+            //Create new stock object with parsed data
+            let newStock = Stock(name: name, symbol: symbol, price: Double(price)!, netChange: changeInPriceStringClean.doubleValue, netChangeInPercentage: changeInPercentStringClean.doubleValue)
+            
+            //If stock isn't already in the stock array, then add it
+            if !stockArray.contains(newStock){
+                stockArray.append(newStock)
+            }
+        }
+        
+        //Update the data in our WatchList
+        WatchListManager.sharedInstance.updateStockArrayWithNewData(stockArray)
+        
+        //Send out update notification
+        notififyListenersOfUpdates()
+    }
+    
+    /*  
+        Will post a notification that lets any listening View Controllers
+        know that updates have been received.
+     */
+    func notififyListenersOfUpdates() {
+        //mark request as finished
+        isMakingRequest = false
+        
+        dispatch_async(dispatch_get_main_queue(), {
+            //Inform listeners that updates have been received and parsed
+            if !self.shouldCancelUpdate {
+                self.notificationCenter.postNotificationName(Constants.kNotificationStockPricesUpdated, object: nil, userInfo: nil)
```

Comment:
`kNotificationStockPricesUpdated` constants with a `k` is not a cocoa or swift thing, if you look at the naming of the classes of other notifications in the framework, this should be 

`StockPricesUpdatedNotification`